### PR TITLE
feat: Add User Task Restrictions on SaaS Document

### PR DESCRIPTION
**Pull request by nathansandi**
## Description

Add the Access control session mentioning 
- Groups
- User Task Restrictions

## When should this change go live?

8.5.0-alpha2 

- [x] This change is not yet live and should not be merged until 12/03/2024 (apply `hold` label or convert to draft PR)?
## PR Checklist

- [x] I have added changes to the relevant `/versioned_docs` directory, or they are not for an **already released version**.
- [x] I have added changes to the main `/docs` directory (aka `/next/`), or they are not for **future versions**.
- [ ] My changes require an [Engineering review](https://github.com/camunda/camunda-docs/blob/main/howtos/documentation-guidelines.md#review-process), and I've assigned an engineering manager or tech lead as a reviewer, or my changes do not require an Engineering review.
- [ ] My changes require a [technical writer review](https://github.com/camunda/camunda-docs/blob/main/howtos/documentation-guidelines.md#review-process), and I've assigned @christinaausley as a reviewer, or my changes do not require a technical writer review.

### CODE REVIEW

File: docs/components/concepts/access-control/user-groups.md [hunk 1]
```diff
@@ -0,0 +1,28 @@
...
+description: "Control the level of access a user or group has to perform tasks in the system via user task access restrictions."
+---
+
+User groups are a way to organize users in the system. They can be used to control the level of access a user or group has to perform tasks in the system via [user task restrictions](user-task-access-restrictions.md).
+
+### Creating a group
+
+To create a group, navigate to the **Organization** section of Console and click on the **Groups** tab.
+
+![Groups Management](../assets/access-control/group-management.png)
```
Comment (christinaausley) [inline docs/components/concepts/access-control/user-groups.md:14]: Note that if a user hasn't created a group before, this screenshot will look different. I've adjusted the UI references below as such.

File: docs/components/concepts/access-control/user-groups.md [hunk 2]
```diff
@@ -0,0 +1,28 @@
...
+
+![Create a group](../assets/access-control/create-group.png)
+
+#### Adding users to a group
+
+To add users to a group, navigate to the **Organization** section of Console and click on the **Users** tab.
+
+![Groups Members](../assets/access-control/group-members.png)
+
+Select the user you want to add to a group and click **Assign**. Select the group you want to add the user to and click **Assign members**.
```
Comment (christinaausley) [inline docs/components/concepts/access-control/user-groups.md:26]: Are both of these actions required? I just clicked on the group and **Assign members**.

File: docs/components/concepts/access-control/user-groups.md [hunk 3]
```diff
@@ -0,0 +1,28 @@
...
+
+![Assign a Member](../assets/access-control/create-group.png)
```
Comment (christinaausley) [inline docs/components/concepts/access-control/user-groups.md:28]: I think this is meant to be the assign-member screenshot.

File: docs/components/concepts/access-control/user-groups.md (hunk 2 above)
Comment (nathansandi) [reply docs/components/concepts/access-control/user-groups.md]: I just fixed, :) The part of assign members should be in another paragraph.

**Review by christinaausley — DISMISSED** [review]
Thanks for addressing my feedback!

File: docs/components/concepts/access-control/user-groups.md [hunk 4]
```diff
@@ -0,0 +1,28 @@
+---
+id: user-groups
+title: "User groups"
+sidebar_label: "User groups"
+description: "Control the level of access a user or group has to perform tasks in the system via user task access restrictions."
+---
+
+User groups are a way to organize users in the system. They can be used to control the level of access a user or group has to perform tasks in the system via [user task restrictions](user-task-access-restrictions.md).
```
Comment (christian-konrad) [inline docs/components/concepts/access-control/user-groups.md:8]:
```suggestion
User groups are a way to organize users in the system. Use them to manage access to user tasks via [user task restrictions](user-task-access-restrictions.md).
```

Shorter, action-oriented wording

File: docs/components/concepts/access-control/user-groups.md [hunk 5]
```diff
@@ -0,0 +1,28 @@
...
+
+### Creating a group
+
+To create a group, navigate to the **Organization** section of Console and click on the **Groups** tab.
+
+![Groups Management](../assets/access-control/group-management.png)
+
+Click **Create a group** and fill in the **Name** field with your group name.
```
Comment (christian-konrad) [inline docs/components/concepts/access-control/user-groups.md:16]:
```suggestion
Click **Create a group** and enter the name of the group.
```

File: docs/components/concepts/access-control/user-task-access-restrictions.md [hunk 6]
```diff
@@ -0,0 +1,30 @@
...
+title: "User task access restrictions"
+sidebar_label: "User task access restrictions"
+description: "Control the level of access a user or group has to perform tasks in the system via user task access restrictions."
+---
+
+:::caution
+User task access restrictions are enabled by default on SaaS.
+:::
+
+User task access restrictions allow you to control the level of access a [user](../../console/manage-organization/manage-users.md) or
```
Comment (christian-konrad) [inline docs/components/concepts/access-control/user-task-access-restrictions.md:12]:
```suggestion
User task access restrictions allow you to restrict access of user tasks in Tasklist to [users](../../console/manage-organization/manage-users.md) or
```

File: docs/components/concepts/access-control/user-task-access-restrictions.md [hunk 7]
```diff
@@ -0,0 +1,30 @@
...
+[group](user-groups.md) has to perform BPMN user tasks where they are candidates.
```
Comment (christian-konrad) [inline docs/components/concepts/access-control/user-task-access-restrictions.md:13]:
```suggestion
[groups](user-groups.md) where they are assignees or candidates.
```

**Review by christian-konrad — CHANGES_REQUESTED** [review]
Some small changes, but nothing significant. Thanks @nathansandi

**Comment by nathansandi** [comment]
Thanks @christian-konrad  and @christian-konrad  for the reviews
I addressed all changes.

**Review by christinaausley — APPROVED** [review]

**Pull request merged**